pkg/services/store/entity/db/dbimpl/dbEngine.go

```diff
@@ -1,75 +1,98 @@
 package dbimpl
 
 import (
+	"cmp"
 	"fmt"
 	"strings"
 	"time"
 
-	"github.com/grafana/grafana/pkg/infra/tracing"
-	"github.com/grafana/grafana/pkg/services/sqlstore"
-	"github.com/grafana/grafana/pkg/setting"
-	"github.com/grafana/grafana/pkg/util"
+	"github.com/go-sql-driver/mysql"
 	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/store/entity/db"
 )
 
-func getEngineMySQL(cfgSection *setting.DynamicSection, tracer tracing.Tracer) (*xorm.Engine, error) {
-	dbHost := cfgSection.Key("db_host").MustString("")
-	dbName := cfgSection.Key("db_name").MustString("")
-	dbUser := cfgSection.Key("db_user").MustString("")
-	dbPass := cfgSection.Key("db_pass").MustString("")
+func getEngineMySQL(getter *sectionGetter, _ tracing.Tracer) (*xorm.Engine, error) {
+	config := mysql.NewConfig()
+	config.User = getter.String("db_user")
+	config.Passwd = getter.String("db_pass")
+	config.Net = "tcp"
+	config.Addr = getter.String("db_host")
+	config.DBName = getter.String("db_name")
+	config.Params = map[string]string{
+		// See: https://dev.mysql.com/doc/refman/en/sql-mode.html
+		"@@SESSION.sql_mode": "ANSI",
+	}
+	config.Collation = "utf8mb4_unicode_ci"
+	config.Loc = time.UTC
+	config.ServerPubKey = getter.String("db_server_pub_key")
+	config.TLSConfig = getter.String("db_tls_config_name")
+	config.AllowNativePasswords = true
+	config.CheckConnLiveness = true
+	config.ClientFoundRows = true
 
-	// TODO: support all mysql connection options
-	protocol := "tcp"
-	if strings.HasPrefix(dbHost, "/") {
-		protocol = "unix"
+	if err := getter.Err(); err != nil {
+		return nil, fmt.Errorf("config error: %w", err)
 	}
 
-	connectionString := connectionStringMySQL(dbUser, dbPass, protocol, dbHost, dbName)
+	if strings.HasPrefix(config.Addr, "/") {
+		config.Net = "unix"
+	}
 
-	driverName := sqlstore.WrapDatabaseDriverWithHooks("mysql", tracer)
-	engine, err := xorm.NewEngine(driverName, connectionString)
+	// FIXME: get rid of xorm
+	engine, err := xorm.NewEngine(db.DriverMySQL, config.FormatDSN())
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("open database: %w", err)
 	}
 
 	engine.SetMaxOpenConns(0)
 	engine.SetMaxIdleConns(2)
-	engine.SetConnMaxLifetime(time.Second * time.Duration(14400))
+	engine.SetConnMaxLifetime(4 * time.Hour)
 
 	return engine, nil
 }
 
-func getEnginePostgres(cfgSection *setting.DynamicSection, tracer tracing.Tracer) (*xorm.Engine, error) {
-	dbHost := cfgSection.Key("db_host").MustString("")
-	dbName := cfgSection.Key("db_name").MustString("")
-	dbUser := cfgSection.Key("db_user").MustString("")
-	dbPass := cfgSection.Key("db_pass").MustString("")
+func getEnginePostgres(getter *sectionGetter, _ tracing.Tracer) (*xorm.Engine, error) {
```

Comment:
changing this makes me a bit nervious. Could we keep this in a different PR ?

Reply:
The current implementation is broken and won't work, it only works if you use `grafana` / `grafana` (or `grafanatest`) as user and pass. Try adding a space in your password, which broke my local tests.

Reply:
Reduced the items being set to the bare minimum, moved to TODO comments the rest, with documentation references.